src/nodes.js

```diff
@@ -143,10 +143,55 @@ var registerChild = function(parent, key, child) {
 };
 
 
+/**
+ * Cleans a key map, nulling any nodes that are no longer in the DOM.
+ * @param {!Node} node
+ */
+var cleanKeyMap = function(node) {
+  var data = getData(node);
+  var keyMap = data.keyMap;
+  
+  if (!keyMap) {
+    return;
+  }
+
+  for (var key in keyMap) {
+    if (!keyMap[key].parentNode) {
+      delete keyMap[key];
+    }
+  }
+
+  data.keyMapValid = true;
+};
+
+
+/**
+ * Marks the key map for a given Element as invalid. It will be cleaned the
+ * next time it is needed.
+ * @param {!Node} node
+ */
+var invalidateKeyMap = function(node) {
+  getData(node).keyMapValid = false;
+};
+
+
+/**
+ * @param {!Node} node
+ * @return {boolean} True if the keyMap for the given Node is valid, false
+ *    otherwise.
+ */
+var isKeyMapValid = function(node) {
```

Comment:
The original reason for the `NodeData` getters was that the `NodeData` was null for `Text` nodes. I decided to just initialize a `NodeData` for all nodes in order to allow patching `DocumentFragment`s and simply the implementation, but forgot to remove the getters.

It feels a bit odd to directly access things related to the keyMap in some cases and not in others. I think I am going to change it to explicitly 'import' the patch subtree (like in #86) and create the key maps up front  in a later PR. Then, I will just directly access the key map stuff where needed.